assemblyline/templates/ingress.yaml

```diff
@@ -41,38 +41,94 @@ spec:
                 name: frontend
                 port:
                   number: 3000
+
+---
+
+# SOCKETIO INGRESS (Internal)
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: {{ .Release.Name }}-socketio-ingress
+  annotations:
+    {{- range $key, $value := .Values.ingressAnnotations }}
+    {{ $key }}: {{ $value | quote }}
+    {{- end }}
+    appgw.ingress.kubernetes.io/backend-hostname: "socketio"
+spec:
+  tls:
+    - hosts: [{{ .Values.configuration.ui.fqdn }}]
+      secretName: {{ if .Values.tlsSecretName }} {{.Values.tlsSecretName }} {{else}} {{ .Release.Name }}-generated-cert {{end}}
+  rules:
+    - host: {{ .Values.configuration.ui.fqdn | quote }}
+      http:
+        paths:
           - path: /socket.io/
             pathType: Prefix
             backend:
               service:
                 name: socketio
                 port:
                   number: 5002
+
+---
+
+# UI INGRESS (Internal)
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: {{ .Release.Name }}-ui-ingress
+  annotations:
+    {{- range $key, $value := .Values.ingressAnnotations }}
+    {{ $key }}: {{ $value | quote }}
+    {{- end }}
+    appgw.ingress.kubernetes.io/backend-hostname: "ui"
+spec:
+  tls:
+    - hosts: [{{ .Values.configuration.ui.fqdn }}]
+      secretName: {{ if .Values.tlsSecretName }} {{.Values.tlsSecretName }} {{else}} {{ .Release.Name }}-generated-cert {{end}}
+  rules:
+    - host: {{ .Values.configuration.ui.fqdn | quote }}
+      http:
+        paths:
           - path: /api/
             pathType: Prefix
             backend:
               service:
                 name: ui
                 port:
                   number: 5000
-          {{ if .Values.separateIngestAPI }}
+
+---
+
+# UI-INGEST INGRESS (Internal)
+{{ if .Values.separateIngestAPI }}
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: {{ .Release.Name }}-ui-ingest-ingress
+  annotations:
+    {{- range $key, $value := .Values.ingressAnnotations }}
+    {{ $key }}: {{ $value | quote }}
+    {{- end }}
+    appgw.ingress.kubernetes.io/backend-hostname: "ui"
+spec:
+  tls:
+    - hosts: [{{ .Values.configuration.ui.fqdn }}]
+      secretName: {{ if .Values.tlsSecretName }} {{.Values.tlsSecretName }} {{else}} {{ .Release.Name }}-generated-cert {{end}}
+  rules:
+    - host: {{ .Values.configuration.ui.fqdn | quote }}
+      http:
+        paths:
           - path: /api/v4/ingest/
             pathType: Prefix
             backend:
               service:
                 name: ui-ingest
                 port:
                   number: 5000
-          {{end}}
-          {{ if .Values.internalELKStack}}
```

Comment:
We'd probably want to create an ingress for Kibana if the user enables it in the chart.

Reply:
I'll add it in